azure-pipelines-dependency-graph-testing.yml

```diff
@@ -178,158 +182,214 @@ stages:
-      #         pip install .
-      #       displayName: 'Install dependencies'
-
-      #     - script: |
-      #         pip install --requirement requirements.txt
-      #         pip install pytest pytest-cov pytest-azurepipelines
-      #         python scripts/determine_tests_to_run.py --depth 3 | xargs pytest --mysql --no-postgresql --no-spark --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html --ignore=tests/cli --ignore=tests/integration/usage_statistics
-      #       displayName: 'pytest'
-
-      # - job: mssql
-      #   condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true), eq(variables.isMain, true))
-
-      #   services:
-      #     mssql: mssql
-
-      #   variables:
-      #     python.version: '3.8'
-
-      #   steps:
-      #     - task: UsePythonVersion@0
-      #       inputs:
-      #         versionSpec: '$(python.version)'
-      #       displayName: 'Use Python $(python.version)'
-
-      #     - bash: python -m pip install --upgrade pip  # pip==20.2.4
-      #       displayName: 'Update pip'
-
-      #     - script: |
-      #         sqlcmd -U sa -P "ReallyStrongPwd1234%^&*" -Q "CREATE DATABASE test_ci;" -o create_db_output.txt
-
-      #     - script: |
-      #         pip install --requirement requirements-dev-base.txt --requirement requirements-dev-sqlalchemy.txt --constraint constraints-dev.txt
-      #         pip install --requirement requirements.txt
-      #         pip install .
-      #       displayName: 'Install dependencies'
-
-      #     - script: |
-      #         pip install pytest pytest-cov pytest-azurepipelines
-      #         python scripts/determine_tests_to_run.py --depth 3 | xargs pytest --mssql --no-postgresql --no-spark --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html --ignore=tests/cli --ignore=tests/integration/usage_statistics
-      #       displayName: 'pytest'
+      - job: comprehensive
+        condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true), eq(variables.isMain, true))
+
+        services:
+          postgres: postgres
+
+        variables:
+          GE_pytest_opts: ''
+
+        strategy:
+          matrix:
+            # Python36:
+            #   python.version: '3.6'
+            #   pandas.version: 'latest'
+            #   GE_pytest_pip_opts: '"pyspark<3.0.0" --requirement requirements-dev.txt --constraint constraints-dev.txt'
+            # Python37:
+            #   python.version: '3.7'
+            #   pandas.version: 'latest'
+            #   GE_pytest_pip_opts: '"pyspark<3.0.0" --requirement requirements-dev.txt --constraint constraints-dev.txt'
+            Python38:
+              python.version: '3.8'
+              pandas.version: 'latest'
+              GE_pytest_pip_opts: '--requirement requirements-dev.txt --constraint constraints-dev.txt'
```

Comment:
Went back and forth on this a bit but decided to only test `3.8`. Each version takes up a slot and we only have 10. This testing shouldn't be a bottleneck for actual development so I'm narrowing the testing strategy.